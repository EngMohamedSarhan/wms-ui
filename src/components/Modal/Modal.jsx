import Dialog from "@mui/material/Dialog";
import DialogActions from "@mui/material/DialogActions";
import DialogContent from "@mui/material/DialogContent";
import PropTypes from "prop-types";
import React from "react";

import ModalTitle from "../ModalTitle/ModalTitle";

const Modal = ({
  isOpen,
  title,
  onClose,
  children,
  buttons,
  titleProps,
  ...props
}) => (
  <Dialog
    onClose={onClose}
    aria-labelledby="notification-modal"
    open={isOpen}
    {...props}
  >
    <ModalTitle title={title} onClose={onClose} {...titleProps} />
    <DialogContent sx={{ maxHeight: 300, minWidth: 300 }}>{children}</DialogContent>
    <DialogActions>{buttons}</DialogActions>
  </Dialog>
);

Modal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  title: PropTypes.string,
  children: PropTypes.node,
  buttons: PropTypes.node,
  titleProps: PropTypes.object,
};

export default Modal;
