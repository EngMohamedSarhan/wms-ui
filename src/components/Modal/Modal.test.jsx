/* eslint-disable no-undef */
import { render } from "@testing-library/react";
import React from "react";
import ShallowRenderer from "react-shallow-renderer";

import Modal from "./Modal";

describe("Modal Component", () =>
  test("Modal not rendering", () => {
    const onClose = jest.fn();
    const { queryByTestId } = render(<Modal isOpen={false} onClose={onClose} />);
    const modal = queryByTestId("modal");
    expect(modal).toBeNull();
  }));

describe("Modal Component", () =>
  test("Modal rendering", () => {
    const onClose = jest.fn();
    const { queryByTestId } = render(<Modal isOpen={true} onClose={onClose} />);
    const modal = queryByTestId("modal");
    expect(modal).toBeInTheDocument();
  }));

describe("Modal Component", () =>
  test("Modal snapshot", () => {
    const onClose = jest.fn();
    const renderer = new ShallowRenderer();
    renderer.render(<Modal isOpen={true} onClose={onClose} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  }));
