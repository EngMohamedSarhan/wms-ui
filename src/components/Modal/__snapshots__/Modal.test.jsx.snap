// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component Modal snapshot 1`] = `
<ForwardRef(Dialog)
  aria-labelledby="modal"
  data-testid="modal"
  onClose={[MockFunction]}
  open={true}
>
  <ModalTitle
    onClose={[MockFunction]}
  />
  <ForwardRef(DialogContent)
    sx={
      Object {
        "maxHeight": 300,
        "minWidth": 300,
      }
    }
  />
  <ForwardRef(DialogActions) />
</ForwardRef(Dialog)>
`;

exports[`Modal snapshot Modal snapshot 1`] = `
<ForwardRef(Dialog)
  aria-labelledby="modal"
  data-testid="modal"
  onClose={[MockFunction]}
  open={true}
>
  <ModalTitle
    onClose={[MockFunction]}
  />
  <ForwardRef(DialogContent)
    sx={
      Object {
        "maxHeight": 300,
        "minWidth": 300,
      }
    }
  />
  <ForwardRef(DialogActions) />
</ForwardRef(Dialog)>
`;
